src/handlers/assign.rs

```diff
@@ -963,6 +1008,12 @@ async fn candidate_reviewers_from_names<'a>(
                 Some(FindReviewerError::ReviewerAlreadyAssigned {
                     username: candidate.clone(),
                 })
+            } else if expansion_happend && is_previously_assigned {
```

Comment:
You don't need to check `expansion_happened` here. We should decide per-reviewer basis if they were expanded or not. It's enough to check `reviewer_candidate.origin` here.